File: PYTHON/SmartPowerSupply.py
from CoolingSystem import CoolingSystem
from PowerSupply import PowerSupply

class SmartPowerSupply(PowerSupply, CoolingSystem):
    def __init__(self, watt, merk, nama):
        super().__init__(watt, merk, nama)

    def energy_saving_mode(self):
        print("Mengaktifkan mode hemat energi pada power supply...")

File: CPP/SmartPowerSupply.cpp
#pragma once
#include "PowerSupply.cpp"
#include "CoolingSystem.cpp"

class SmartPowerSupply : public PowerSupply, public CoolingSystem {
public:
    SmartPowerSupply(int watt, string merk, string nama)
        : PowerSupply(watt, merk, nama) {}

    void energySavingMode() {
        cout << "Mengaktifkan mode hemat energi pada power supply..." << endl;
    }
};
